mlflow/store/rest_store.py

```diff
@@ -38,27 +39,37 @@ def _api_method_to_info():
 _METHOD_TO_INFO = _api_method_to_info()
 
 
+class RestException(Exception):
+    """Exception thrown on 400-level errors from the REST API"""
+    def __init__(self, json):
+        message = json['error_code']
+        if 'message' in json:
+            message = "%s: %s" % (message, json['message'])
+        super(RestException, self).__init__(message)
+        self.json = json
+
+
 class RestStore(AbstractStore):
     """ Client for a remote tracking server accessed via REST API calls """
 
-    def __init__(self, hostname):
+    def __init__(self, http_request_params):
```

Comment:
Nit: maybe document `http_request_params` / consider renaming it to `http_request_kwargs`? It's currently a little confusing to distinguish from query params.